### PR TITLE
support ! on source and destination parameters to invert the selection

### PR DESCRIPTION
**Pull request by sxpert**

### CODE REVIEW

**Comment by gregdek** [comment]
Thanks @sxpert. @LinusU please review according to guidelines (http://docs.ansible.com/ansible/developing_modules.html#module-checklist) and comment with text 'shipit' or 'needs_revision' as appropriate.

**Comment by LinusU** [comment]
Could you clarify one thing for me, how is the target command supposed to look like? Shouldn't it be `-s !127.0.0.1`?

**Comment by sxpert** [comment]
no, the manual for iptables states (for instance in the source case) : 

> [!] -s, --source address[/mask][,...]
> Source specification. Address can be either a network name, a hostname, a network IP address (with /mask), or a plain IP address. Hostnames will be resolved once only, before the rule is submitted to the kernel.  Please note that  specifying  any  name  to  be resolved  with  a  remote  query  such as DNS is a really bad idea.  The mask can be either an ipv4 network mask (for iptables) or a plain number, specifying the number of 1's at the left side of the network mask.  Thus, an iptables mask of 24 is equivalent to 255.255.255.0.  A "!" argument before the address specification inverts the sense of the address. The flag --src is an alias for this option.  Multiple addresses can be specified, but this will expand to multiple rules (when adding with -A), or will cause multiple rules to be deleted (with -D).

File: system/iptables.py [hunk 1]
```diff
@@ -250,8 +243,16 @@ def append_match(rule, param, match):
 def construct_rule(params):
     rule = []
     append_param(rule, params['protocol'], '-p', False)
-    append_param(rule, params['source'], '-s', False)
-    append_param(rule, params['destination'], '-d', False)
+    source = params['source']
+    if (source is not None) and (len(source)>0) and (source[0]=='!') :
```
Comment (LinusU) [inline system/iptables.py]: No space before `:`, please follow PEP8

**Comment by gregdek** [comment]
Thanks @sxpert for this PR. Unfortunately, it is not mergeable in its current state due to merge conflicts. Please rebase your PR. When you are done, please comment with text 'ready_for_review' and we will put this PR back into review.

**Comment by sxpert** [comment]
@gregdek could you help me with this (am new at this), I can't seem to find the alluded to conflicts

**Comment by sxpert** [comment]
ready_for_review

**Comment by LinusU** [comment]
> ready_for_review

Your pull request contains 127 commits and touches 49 files. Something is going wrong with your rebase...

**Comment by sxpert** [comment]
@LinusU yeah, it seems like so...
can't seem to get this to work ;(

**Comment by sxpert** [comment]
attempt to redo it all

**Pull request closed without merging**